lib/streamlit/delta_generator.py

```diff
@@ -705,8 +705,10 @@ def _prep_data_for_add_rows(
     add_rows_metadata: AddRowsMetadata | None,
 ) -> tuple[Data, AddRowsMetadata | None]:
     if not add_rows_metadata:
-        # When calling add_rows on st.table or st.dataframe we want styles to pass through.
-        return dataframe_util.convert_anything_to_df(data, allow_styler=True), None
+        if dataframe_util.is_pandas_styler(data):
+            # When calling add_rows on st.table or st.dataframe we want styles to pass through.
+            return data, None
```

Comment:
Do I understand correctly that in this case, `data` itself is a Styler and could be sent as is without converting to a pandas DataFrame?

Reply:
Yep, correct. It will be converted into a pandas dataframe at a later point anyway (within the marshall function):

https://github.com/streamlit/streamlit/blob/3aad5e2a97998f56d4fd936f6d6269960426720d/lib/streamlit/delta_generator.py#L660C21-L660C29